File: sdk/typescript/src/exceptions.ts
import type { ProblemDetails } from "./models.js";

/** Raised when the HUBVERY API returns an RFC 7807 problem+json error. */
export class HubveryAPIError extends Error {
  readonly status: number;
  readonly title: string;
  readonly detail?: string;
  readonly code?: string;
  readonly problem: ProblemDetails;

  constructor(problem: ProblemDetails) {
    super(`${problem.status} ${problem.title}: ${problem.detail ?? ""}`.trim());
    this.name = "HubveryAPIError";
    this.problem = problem;
    this.status = problem.status;
    this.title = problem.title;
    this.detail = problem.detail;
    this.code = problem.code;
  }
}

/** Raised when OAuth2 client credentials cannot be exchanged for a token. */
export class HubveryAuthError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "HubveryAuthError";
  }
}

File: sdk/python/src/hubvery_sdk/exceptions.py
"""Exceptions raised by the HUBVERY SDK."""

from __future__ import annotations

from .models import Error


class HubveryAPIError(Exception):
    """Raised when the HUBVERY API returns an RFC 7807 problem+json error."""

    def __init__(self, error: Error) -> None:
        self.error = error
        self.status = error.status
        self.title = error.title
        self.detail = error.detail
        self.code = error.code
        super().__init__(f"{error.status} {error.title}: {error.detail or ''}".strip())


class HubveryAuthError(Exception):
    """Raised when OAuth2 client credentials cannot be exchanged for a token."""
